src/components/data/QueryAuth0.js

```diff
@@ -21,15 +21,35 @@ import { finishAuthenticationFlow } from 'console/utils/auth0';
 export default class QueryAuth0 extends React.PureComponent {
   static propTypes = {
     accessToken: PropTypes.string,
-    fetchUserProfile: PropTypes.func.isRequired,
+    userProfileReceived: PropTypes.func.isRequired,
     history: PropTypes.object.isRequired,
     loginFailed: PropTypes.func.isRequired,
     logUserIn: PropTypes.func.isRequired,
   };
 
+  // The motivation for using componentWILLMount instead of the more regular componentDidMount
```

Comment:
I don't know that we need to include this explanation. I'm not a huge fan of peppering prose in the code unless it provides vitally important information.

Reply:
`componentWillMount` is going to be removed soon: https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html

The blog post goes into a lot more detail, but suggests this:

> When supporting server rendering, it’s currently necessary to provide the data synchronously – componentWillMount was often used for this purpose but the constructor can be used as a replacement.

Reply:
good to know! 

i don't think it's a big deal if we see a small flash of a login button. currently due to the fetch it can linger for a couple seconds depending on connection speed. so long as that's not the case i'm not too concerned.

Reply:
however perhaps with other query components we will care about synchronous operations